src/code/z_sram.c

```diff
@@ -505,77 +502,69 @@ void Sram_WriteSave(SramContext* sramCtx) {
 
-    dayTime = gSaveContext.dayTime;
+    dayTime = ((void)0, gSaveContext.dayTime);
 
     for (slotNum = 0; slotNum < 3; slotNum++) {
-        offset = gSramSlotOffsets[0][slotNum];
+        offset = gSramSlotOffsets[slotNum];
         osSyncPrintf("ぽいんと＝%x(%d)    SAVE_MAX=%d\n", offset, gSaveContext.fileNum, sizeof(Save));
         MemCopy(&gSaveContext, sramCtx->readBuff + offset, sizeof(Save));
 
         oldChecksum = gSaveContext.checksum;
         gSaveContext.checksum = 0;
+        ptr = (u16*)&gSaveContext;
         osSyncPrintf("\n＝＝＝＝＝＝＝＝＝＝＝＝＝  Ｓ（%d） ＝＝＝＝＝＝＝＝＝＝＝＝＝\n", slotNum);
 
-        // j = 0;
-        newChecksum = 0;
-        ptr = (u16*)&gSaveContext;
-        for (i = 0; i < CHECKSUM_SIZE; i++) {
-            j += 2;
+        for (i = newChecksum = j = 0; i < CHECKSUM_SIZE; i++, offset += 2) {
             newChecksum += *ptr++;
         }
 
         // SAVE checksum calculation
         osSyncPrintf("\nＳＡＶＥチェックサム計算  j=%x  mmm=%x  ", newChecksum, oldChecksum);
 
-        if (oldChecksum != newChecksum) {
+        if (newChecksum != oldChecksum) {
             // checksum didnt match, try backup save
-            osSyncPrintf("ＥＲＲＯＲ！！！ ＝ %x(%d)\n", offset, slotNum);
-            offset = gSramSlotOffsets[1][slotNum];
+            osSyncPrintf("ＥＲＲＯＲ！！！ ＝ %x(%d)\n", gSramSlotOffsets[slotNum], slotNum);
+            offset = gSramSlotOffsets[slotNum + 3];
             MemCopy(&gSaveContext, sramCtx->readBuff + offset, sizeof(Save));
 
             oldChecksum = gSaveContext.checksum;
             gSaveContext.checksum = 0;
+            ptr = (u16*)&gSaveContext;
             osSyncPrintf("================= ＢＡＣＫ─ＵＰ ========================\n");
 
-            newChecksum = 0;
-            ptr = (u16*)&gSaveContext;
-            for (i = 0; i < CHECKSUM_SIZE; i++) {
-                j += 2;
+            for (i = newChecksum = j = 0; i < CHECKSUM_SIZE; i++, offset += 2) {
                 newChecksum += *ptr++;
             }
             // (B) SAVE checksum calculation
             osSyncPrintf("\n（Ｂ）ＳＡＶＥチェックサム計算  j=%x  mmm=%x  ", newChecksum, oldChecksum);
 
-            if (oldChecksum != newChecksum) {
+            if (newChecksum != oldChecksum) {
                 // backup save didnt work, make new save
-                // offPtr = &gSramSlotOffsets[1][slotNum];
-                osSyncPrintf("ＥＲＲＯＲ！！！ ＝ %x(%d+3)\n", offset, slotNum);
+                osSyncPrintf("ＥＲＲＯＲ！！！ ＝ %x(%d+3)\n", gSramSlotOffsets[slotNum + 3], slotNum);
                 bzero(&gSaveContext.entranceIndex, sizeof(gSaveContext.entranceIndex));
                 bzero(&gSaveContext.linkAge, sizeof(gSaveContext.linkAge));
                 bzero(&gSaveContext.cutsceneIndex, sizeof(gSaveContext.cutsceneIndex));
-                bzero(&gSaveContext.dayTime, sizeof(gSaveContext.dayTime));
+                //! @bug: The size here was probably not intended to be multiplied by 2
+                bzero(&gSaveContext.dayTime, sizeof(gSaveContext.dayTime) * 2);
```

Comment:
I think all these should be `sizeof(s32)` since this indicates they clearly didn't use sizeof on the actual variable. It's not really a bug either imo, since it just zeroes out words in the struct and alignment ensures it won't overwrite something unintended.